scripts/update-esbuild-versions.js

```diff
@@ -0,0 +1,81 @@
+const https = require("https");
+const { exec } = require('shelljs');
+const { mkdirSync, rmdirSync, createWriteStream } = require('fs');
+const { join } = require('path');
+const { tmpdir } = require('os');
+
+const PLATFORMS = {
+  "esbuild_darwin": "esbuild-darwin-64",
+  "esbuild_windows": "esbuild-windows-64",
+  "esbuild_linux": "esbuild-linux-64"
+}
+
+function getUrlAsString(url) {
+  return new Promise((resolve, reject) => {
+    https.get(url, (res) => {
+      if (res.statusCode !== 200) {
+        console.error(res);
+        return reject();
+      }
+      
+      let body = '';
+      res.on("data", (chunk) => body += chunk);
+      res.on("end", () => resolve(String(body)));
+    });
+  });
+}
+
+function downloadFile(url, dest) {
+  return new Promise((resolve, reject) => {
+    const file = createWriteStream(dest);
+
+    const request = https.get(url, (response) => {
+        response.pipe(file);
+    });
+
+    file.on('finish', () => {
+      file.end();
+      resolve();
+    });
+  });
+};
+
+async function main() {
+  console.log('""" Generated code; do not edit\nUpdate by running yarn update-esbuild-versions\n\nHelper macro for fetching esbuild versions for internal tests and examples in rules_nodejs\n"""\n');
+  console.log('load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")\n')
+
+  const latestVersion = JSON.parse(await getUrlAsString('https://registry.npmjs.org/esbuild/latest')).version;
```

Comment:
Perhaps move the `JSON.parse` into the `getUrlAsString` method and refactor it to be called `fetch`? It's the only placed its used, so doing the parse outside seems like it's implying in other places it's needed as a string.